selection.go

```diff
@@ -107,6 +96,7 @@ func (s *controllingSelector) HandleSucessResponse(m *stun.Message, local, remot
 	p := s.agent.addValidPair(local, remote)
 
 	if pendingRequest.isUseCandidate {
+		s.nominatedPair = nil
```

Comment:
why setting it to nil? Remember we cannot nominate two different pairs

Reply:
If the`validPair` times out we might need to nominate a different pair. We can move this logic/setting to `validateSelectedPair` instead though!

Reply:
I don't think so, from the RFC:

```
https://tools.ietf.org/html/rfc8445#section-8.1.1
  The criteria for stopping the connectivity checks and for picking a
   pair for nomination are outside the scope of this specification.
   They are a matter of local optimization.  The only requirement is
   that the agent MUST eventually pick one and only one candidate pair
   and generate a check for that pair with the USE-CANDIDATE attribute
   set.

   Once the controlling agent has successfully nominated a candidate
   pair (Section 7.2.5.3.4), the agent MUST NOT nominate another pair
   for same component of the data stream within the ICE session.  Doing
   so requires an ICE restart.
```

Reply:
Aye nice check, mind killing that line?